tests/lib.rs

```diff
@@ -1,726 +1,935 @@
-extern crate netcdf;
+#![cfg(test)]
 
-extern crate ndarray;
-use ndarray::ArrayD;
-use netcdf::{test_file, test_file_new};
+/// Get location of the test files
+fn test_location() -> std::path::PathBuf {
+    use std::path::Path;
 
-// Failure tests
-#[test]
-#[should_panic(expected = "No such file or directory")]
-fn bad_filename() {
-    let f = test_file("blah_stuff.nc");
-    let _file = netcdf::open(&f).unwrap();
+    let mnf_dir = std::env::var("CARGO_MANIFEST_DIR").unwrap();
+    Path::new(&mnf_dir).join("tests").join("testdata")
 }
 
-// Read tests
 #[test]
-fn root_dims() {
-    let f = test_file("simple_xy.nc");
+/// Use a path to open the netcdf file
+fn use_path_to_open() {
+    let path = test_location().join("simple_xy.nc");
 
-    let file = netcdf::open(&f).unwrap();
-    assert_eq!(f, file.name);
+    let _file = netcdf::File::open(path).unwrap();
+}
 
-    assert_eq!(file.root.dimensions.get("x").unwrap().len, 6);
-    assert_eq!(file.root.dimensions.get("y").unwrap().len, 12);
+#[test]
+/// Use a string to open
+fn use_string_to_open() {
+    let f: String = test_location()
+        .join("simple_xy.nc")
+        .to_str()
+        .unwrap()
+        .to_string();
+    let _file = netcdf::open(f).unwrap();
 }
 
+// Failure tests
 #[test]
-fn global_attrs() {
-    let f = test_file("patmosx_v05r03-preliminary_NOAA-19_asc_d20130630_c20140325.nc");
+fn bad_filename() {
+    let f = test_location().join("blah_stuff.nc");
+    let res_file = netcdf::File::open(&f);
+    assert_eq!(res_file.unwrap_err(), netcdf::error::Error::Netcdf(2));
+}
 
-    let file = netcdf::open(&f).unwrap();
-    assert_eq!(f, file.name);
+// Read tests
+#[test]
+fn root_dims() {
+    let f = test_location().join("simple_xy.nc");
 
-    let ch1_attr = file.root.attributes.get("CH1_DARK_COUNT").unwrap();
-    let ch1 = ch1_attr.get_float(false).unwrap();
-    let eps = 1e-6;
-    assert!((ch1-40.65863).abs() < eps);
-    let ch1 = ch1_attr.get_int(true).unwrap();
-    assert_eq!(ch1, 40);
+    let file = netcdf::File::open(&f).unwrap();
+    assert_eq!("simple_xy.nc", file.name());
 
-    let sensor_attr = file.root.attributes.get("sensor").unwrap();
-    let sensor_data = sensor_attr.get_char(false).unwrap();
-    assert_eq!("AVHRR/3".to_string(), sensor_data);
+    assert_eq!(file.root().dimensions().get("x").unwrap().len(), 6);
+    assert_eq!(file.root().dimensions().get("y").unwrap().len(), 12);
```

Comment:
`.get("x").unwrap()` makes me think of `.get()` and the matching indexing syntax on a slice - could it be `.dimensions()["x"]` ? Supporting both string and index indexing is feasible.